### DISCUSSION TITLE
Explain "failed to set hostkey preference" error somewhere

### DISCUSSION THREAD

**Discussion by msifd** [jj-vcs/jj#5027] [General]
Hello! Please add somewhere in docs explanation that errors like

```bash
$ jj git clone git@github.com:martinvonz/jj.git
Fetching into new repo in "\\?\D:\Dev\jujutsu\jj"
Error: failed to set hostkey preference: The requested method(s) are not currently supported; class=Ssh (23)
Hint: Jujutsu uses libssh2, which doesn't respect ~/.ssh/config. Does `ssh -F /dev/null` to the host work?
```
(at least on Windows) means you need to fix your `.ssh/known_hosts`. It took me three days and two computers to figure it out.

So the easy way to fix it is to delete/rename current `known_hosts`, clone repo using git and agree to create new `known_hosts` (this time with githubs `ssh-ed25519` key).

The root cause was that libssh2 doesn't support `ecdsa-sha2-nistp256` keys while git works fine.

**Comment by QiangF** [comment]
Is there any other workaround without using git?

**Comment by QiangF** [comment]
Related bug report: https://github.com/jj-vcs/jj/issues/3322
